model/src/main/kotlin/RemoteArtifact.kt

```diff
@@ -39,7 +39,7 @@ data class RemoteArtifact(
          * The name of the algorithm used to calculate the [hash].
          */
         @JsonProperty("hash_algorithm")
-        val hashAlgorithm: String
+        val hashAlgorithm: HashAlgorithm
```

Comment:
I'm concerned that we lose information when we use the enum here, as we limit possible values to known algorithms. E.g. currently if the algorithm was SHA-256, or if MD5 was misspelled as MD-5, we would simply replace this with unknown.
I think we should at least add all algorithms supported by the DigestUtils class to the enum, which are MD2, MD5, SHA1, SHA256, SHA384, and SHA512. This also should be improved in the Main.verifyChecksum method in downloader. Maybe it would also be a good idea to move this method to the new enum?

Reply:
> I'm concerned that we lose information

Good point.

> I think we should at least add all algorithms supported by the DigestUtils class to the enum

Done.

> Maybe it would also be a good idea to move this method to the new enum?

Then the `model` would depend on `org.apache.commons.codec.digest.DigestUtils`. Is that acceptable for us?

Reply:
Not sure, I wanted to do an experiment with the model and Kotlin-JS, so let's postpone moving the method.